hs_core/hydroshare/hs_bagit.py

```diff
@@ -89,21 +91,21 @@ def create_bag_files(resource, fed_zone_home_path=''):
     # to_file_name = '{res_id}/data/visualization/'.format(res_id=resource.short_id)
     # istorage.saveFile('', to_file_name, create_directory=True)
 
-    # create resourcemetadata.xml and upload it to iRODS
-    from_file_name = '{path}/resourcemetadata.xml'.format(path=bagit_path)
+    # create resourcemetadata.xml in local directory and upload it to iRODS
+    from_file_name = os.path.join(temp_path, 'resourcemetadata.xml')
     with open(from_file_name, 'w') as out:
         # resources that don't support file types this would write only resource level metadata
         # resource types that support file types this would write resource level metadata
         # as well as file type metadata
         out.write(resource.get_metadata_xml())
-    to_file_name = '{res_id}/data/resourcemetadata.xml'.format(res_id=resource.short_id)
-    if fed_zone_home_path:
-        to_file_name = '{fed_zone_home_path}/{rel_path}'.format(
-                            fed_zone_home_path=fed_zone_home_path,
-                            rel_path=to_file_name)
+    if resource.resource_federation_path:
+        to_file_name = os.path.join(resource.resource_federation_path, 'data',
```

Comment:
resource.resource_federation_path does not contain short_id, so the logic here is not correct. I think ```root_path``` can be used here to simplify this if..else statement into one statement like ```to_file_name = os.path.join(resource.root_path, 'data', 'resourcemetadata.xml')```